### PR TITLE
OpenShift manifests now reference the correct artifact

### PR DESCRIPTION
**Pull request by iocanel**
Fix for issue #6576 

This pull request ensures that the correct jar is referenced in the openshift manifests.

### CODE REVIEW

**Review by geoand — APPROVED** [review]

**Comment by geoand** [comment]
The CI failure is unrelated to the PR. Hopefully https://github.com/quarkusio/quarkus/pull/6579 will fix it

File: extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java [hunk 1]
```diff
@@ -220,6 +222,8 @@ private Project createProject(ApplicationInfoBuildItem app, ArchiveRootBuildItem
         Project project = FileProjectFactory.create(archiveRootBuildItem.getArchiveLocation().toFile());
         BuildInfo buildInfo = new BuildInfo(app.getName(), app.getVersion(),
                 "jar", project.getBuildInfo().getBuildTool(),
+                archiveRootBuildItem.getArchiveRoot()
+                        .resolve(String.format(OUTPUT_ARTIFACT_FORMAT, app.getName(), app.getVersion())),
```
Comment (gsmet) [inline extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java]: That won't work. You can configure the name and the suffix in `PackageConfig`.

See https://github.com/quarkusio/quarkus/blob/master/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java#L179 for instance.

**Review by gsmet — CHANGES_REQUESTED** [review]

File: extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java (hunk 1 above)
Comment (geoand) [reply extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java]: Aha, nice catch! I didn't even remember that is was configurable

Comment (geoand) [reply extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java]: I think it's fixed now. OK @gsmet?

**Review by gsmet — APPROVED** [review]

**Pull request merged**